qubespdfconverter/server.py

```diff
@@ -185,15 +208,122 @@ def __init__(self, path):
+        - Store the document without additionnal properties [this remove the password]
+        """
+
+        src = "file://"+str(self.path)
+        dst = "file://"+str(self.path)+".nopassword"
+
+        local_context = uno.getComponentContext()
+        resolver = local_context.ServiceManager.createInstanceWithContext(
+            "com.sun.star.bridge.UnoUrlResolver",
+            local_context
+        )
+        ctx = resolver.resolve(
+            "uno:socket,host=localhost,port=2202;urp;StarOffice.ComponentContext"
+        )
+        smgr = ctx.ServiceManager
+        desktop = smgr.createInstanceWithContext("com.sun.star.frame.Desktop", ctx)
+
+        hidden_property = PropertyValue()
+        hidden_property.Name = "Hidden"
+        hidden_property.Value = True
+
+        password_property = PropertyValue()
+        password_property.Name = "Password"
+        password_property.Value = self.password
+
+        document = desktop.loadComponentFromURL(
+            src,
+            "_blank",
+            0,
+            (password_property, hidden_property,)
+        )
+        document.storeAsURL(dst, ())
 
 
     async def sanitize(self):
         """Start sanitization tasks"""
-        self.pagenums = self._pagenums()
-        self.batch = asyncio.Queue(self.pagenums)
 
-        send(self.pagenums)
+        password_success = False
+        mimetype = magic.detect_from_filename(str(self.path)).mime_type
+        if mimetype.startswith("video/") or mimetype.startswith("audio/"):
+            raise ValueError
+        if mimetype.startswith("image/"):
+            self.pagenums = 1
+        else:
+            if mimetype == "application/pdf":
+                while not password_success:
+                    cmd = ["pdfinfo", "-opw", self.password, "-upw", self.password, str(self.path)]
+                    try:
+                        password_success = not b"Incorrect password" in subprocess.\
+                                run(cmd, capture_output=True, check=True).stderr
+                    except subprocess.CalledProcessError:
+                        password_success = False
+                        self._read_password(password_success)
+            else:
+                # Performance could be improved by only starting
+                # the libreoffice when needed (aka: when the file need to be decrypted).
+                # But code is simpler that way
+
+                # Launch libreoffice server
+                cmd = [
+                    "libreoffice",
+                    "--accept=socket,host=localhost,port=2202;urp;",
+                    "--norestore",
+                    "--nologo",
+                    "--nodefault"
+                ]
+                libreoffice_process = subprocess.Popen(cmd, stderr=open(os.devnull, 'wb'))
+
+                # Wait until libreoffice server is ready
+                sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```

Comment:
I think you're missing a `sock.close()`.

Reply:
Also, I think sockets can be used as context managers as well, which would make this cleaner.